typescript/hacker-news-agent/src/workflows.ts

```diff
@@ -0,0 +1,219 @@
+    summary: {
+      total_stories: researchHistory.reduce(
+        (sum, r) => sum + r.stories_found,
+        0,
+      ),
+      total_comments: researchHistory.reduce(
+        (sum, r) => sum + r.comments_analyzed,
+        0,
+      ),
+      queries_executed: researchHistory.map((r) => r.query),
+      avg_relevance:
+        allFindings.length > 0
+          ? allFindings.reduce((sum, f) => sum + (f.relevance_score || 0), 0) /
+            allFindings.length
+          : 0,
+    },
+  };
+}
+
+async function researchQueryWorkflowFunction(
+  topic: string,
+  query: string,
+  iteration: number,
+): Promise<IterationResult> {
+  console.log(`🔍 Searching for stories: '${query}'`);
+
+  // Step 1: Search Hacker News for stories about the topic
+  const stories = await DBOS.runStep(() => searchHackerNews(query, 30), {
+    name: "searchHackerNews",
+  });
+
+  if (stories.length > 0) {
+    console.log(`📚 Found ${stories.length} stories, analyzing all stories...`);
+    stories.forEach((story, i) => {
+      const title = (story.title || "No title").slice(0, 80);
+      const points = story.points || 0;
+      const numComments = story.num_comments || 0;
+      console.log(
+        `  📖 Story ${i + 1}: ${title}... (${points} points, ${numComments} comments)`,
+      );
+    });
+  } else {
+    console.log("❌ No stories found for this query");
+  }
+
+  // Step 2: Gather comments from all stories found
+  const comments: any[] = [];
+  if (stories.length > 0) {
+    console.log(`💬 Reading comments from ALL ${stories.length} stories...`);
+
+    for (let i = 0; i < stories.length; i++) {
+      const story = stories[i];
+      const storyId = story.objectID;
+      const title = (story.title || "Unknown").slice(0, 50);
+      const numComments = story.num_comments || 0;
+
+      if (storyId && numComments > 0) {
+        console.log(
+          `  💭 Reading comments from: ${title}... (${numComments} comments)`,
+        );
+        const storyComments = await DBOS.runStep(
+          () => getComments(storyId, 10),
+          { name: "getComments" },
+        );
+        comments.push(...storyComments);
+        console.log(`    ✓ Read ${storyComments.length} comments`);
+      } else if (storyId) {
+        console.log(`  📖 Story has no comments: ${title}`);
+      } else {
+        console.log(`  ❌ No story ID available for: ${title}`);
+      }
+    }
+  }
+
+  // Step 3: Evaluate gathered data and return findings
+  console.log(
+    `🤔 Analyzing findings from ${stories.length} stories and ${comments.length} comments...`,
+  );
+  const evaluation = await DBOS.runStep(
+    () => evaluateResults(topic, query, stories, comments),
+    { name: "evaluateResults" },
+  );
+
+  return {
+    iteration,
+    query,
+    stories_found: stories.length,
+    comments_analyzed: comments.length,
+    evaluation,
+    stories,
+    comments,
+  };
+}
+
+// Register DBOS workflows
+export const agenticResearchWorkflow = DBOS.registerWorkflow(
```

Comment:
By default, the workflow name is "agenticResearchWorkflowFunction" right? It might be clearer to set the workflow name as "agenticResearchWorkflow" here.